org-code-javabuilder/lib/src/main/java/org/code/javabuilder/JavaRunner.java

```diff
@@ -61,4 +54,43 @@ public void runCode()
       throw new InternalFacingException("Error closing urlClassLoader: " + e, e);
     }
   }
+
+  /**
+   * Finds the main method in the set of files in fileManager if it exists.
+   *
+   * @param urlClassLoader class loader pointing to location of compiled classes
+   * @return the main method if it is found
+   * @throws UserFacingException if there is an issue loading a class
+   * @throws UserInitiatedException if there is more than one main method or no main method
+   */
+  public Method findMainMethod(URLClassLoader urlClassLoader)
```

Comment:
I think there would be value in having unit tests on JavaRunner, but any unit test I attempted ended up being mostly mocking objects and checking I returned the right mock, which did not seem helpful. Any tips on best practices here would be helpful--are unit tests useful for something like this that requires loading already compiled code, or do we need some sort of integration test?

Reply:
Not sure if this is what you're referring to: I could see it being useful to test that this always throws a `UserInitiatedException` if no main is found or if 2 mains are found.

Reply:
I agree, although to write a unit test that does that I have to mock basically everything, which doesn't seem like the most useful test to me.

Reply:
maybe? Part of the problem I'm having is we have multiple related prerequisites for running JavaRunner (classes must be both defined in the fileManager and compiled to the executable location). I can add a note to the existing task to add tests for JavaRunner that we may need to do some refactoring.